vivarium/core/engine.py

```diff
@@ -500,8 +511,10 @@ def calculate_update(
             ``path``) and the store at ``path``.
         """
         store, states = self.process_state(path, process)
-        return self._process_update(
-            path, process, store, states, interval)
+        if process.update_condition(interval, states):
+            return self._process_update(
+                path, process, store, states, interval)
```

Comment:
Is this the right place for `process.update_condition` -- what if a different `Invoke` is used?

Reply:
This is the update condition for derivers, the other one is in the main `update` loop. I believe it is correct? The different invoke should still work from this point, and not be invoked if the condition is not met.

Reply:
Should this `update_condition` come back then?: https://github.com/vivarium-collective/vivarium-core/blob/f68996dc53c41376a2173a8db0b040082a30a120/vivarium/core/engine.py#L124

Reply:
Okay just verified `Invoke` is used inside the `update_condition` block, which is what we want. Also verified that derivers can be parallelized as well.